models/cow_protocol/ethereum/cow_protocol_ethereum_batches.sql

```diff
@@ -0,0 +1,93 @@
+{{  config(
+        alias='batches',
+        materialized='incremental',
+        unique_key = ['tx_hash'],
+        on_schema_change='fail',
```

Comment:
I'm getting the following error after trying to update the model incrementally:
<img width="647" alt="Capture d’écran 2022-09-21 à 09 38 27" src="https://user-images.githubusercontent.com/66172107/191457344-d4f7bbf4-fdf2-4bbf-9720-fd0f50de911f.png">

it seems to be related to the on_schema_change settings here

Reply:
Based on the suggestion names it seems 
- `append_new_columns` seems weird because why would there be new columns if the table structure hasn't changed?
- `sync_all_columns` also seems intense especially if this means the entire table.

Maybe I should be thinking of the term `column` transposed to `row` here because this is a column oriented database, in which case this would mean... "append new rows" which sounds pretty reasonable or "sync all rows" which would also make sense.

The situation we face here is that the trades table doesn't always have the prices until a while later (the sync schedule of `prices.usd`), thus there are batch records being created for which we can not yet evaluate the "batch_value" so we do want sync some of these "columns" (although not necessarily all - because there comes a point when the table contents before a certain block time become static). 

I suppose I need to learn a bit more on how all this works to give the correct answer here (suggestions welcome).

Reply:
Ok - after taking a closer look at the docs https://docs.getdbt.com/docs/building-a-dbt-project/building-models/configuring-incremental-models

I have changed `on_schema_change` to `sync_all_columns` and also noticed that `incremental_strategy` should probably be `delete+insert` instead. 

I hope that `sync_all_columns` isn't to computationally intensive (although I don't expect we will be changing the schema).

Reply:
@bh2smith thanks, will take a look now, I also think sync_all_columns is the way to go (and I don't think it would be too intensive in this particular case but let me check)